frameworks/cassandra/src/main/java/com/mesosphere/sdk/cassandra/scheduler/ZoneValidator.java

```diff
@@ -39,43 +38,48 @@
                     POD_TYPE, TASK_NAME, newConfig));
         }
 
-        String oldEnv = oldTask.get().getCommand().get().getEnvironment().get(DETECT_ZONES_ENV);
-        String newEnv = newTask.get().getCommand().get().getEnvironment().get(DETECT_ZONES_ENV);
+        String oldEnv = oldTask.get()
+                .getCommand().get()
+                .getEnvironment()
+                .get(EnvConstants.PLACEMENT_REFERENCED_ZONE_ENV);
+        String newEnv = newTask.get()
+                .getCommand().get()
+                .getEnvironment()
+                .get(EnvConstants.PLACEMENT_REFERENCED_ZONE_ENV);
 
         return validateTransition(oldEnv, newEnv);
     }
 
     List<ConfigValidationError> validateTransition(String oldEnv, String newEnv) {
-        if (Strings.isBlank(newEnv)) {
-            throw new IllegalArgumentException(String.format(
-                    "%s is not present in the pod=%s task=%s command environment.",
-                    DETECT_ZONES_ENV, POD_TYPE, TASK_NAME));
-        }
+        boolean oldZones = Boolean.valueOf(oldEnv);
+        boolean newZones = Boolean.valueOf(newEnv);
 
-        if (Strings.isBlank(oldEnv)) {
-            if (newEnv.toLowerCase().equals("false")) {
-                return Collections.emptyList();
-            } else if (newEnv.toLowerCase().equals("true")) {
-                ConfigValidationError error = ConfigValidationError.transitionError(
-                        String.format("%s.%s.env.%s", POD_TYPE, TASK_NAME, DETECT_ZONES_ENV),
-                        "null", newEnv,
-                        String.format("Env value %s cannot change from null to false", DETECT_ZONES_ENV));
+        ConfigValidationError error = ConfigValidationError.transitionError(
+                String.format("%s.%s.env.%s", POD_TYPE, TASK_NAME, EnvConstants.PLACEMENT_REFERENCED_ZONE_ENV),
+                oldEnv, newEnv,
+                String.format(
+                        "Env value %s cannot change from %s to %s",
+                        EnvConstants.PLACEMENT_REFERENCED_ZONE_ENV, oldEnv, newEnv));
 
-                return Arrays.asList(error);
-            } else {
-                throw new IllegalStateException(String.format("%s must be either true or false.", DETECT_ZONES_ENV));
-            }
+        // No Zones --> Zones
+        if (!oldZones && newZones) {
```

Comment:
Is this condition not just `XOR`? Do booleans support equality? `if (oldZones != newZones)`?